ci: init

noaccOS opened the pull request:
closes #5 

depends on #28

noaccOS committed:
chore: add dialyxir dependency

Signed-off-by: Francesco Noacco <francesco.noacco@secomind.com>

noaccOS committed:
ci: init

Signed-off-by: Francesco Noacco <francesco.noacco@secomind.com>

rbino commented:
Let's use the standard CI flow here, the rest of us are not Nix-savvy enough to start using this as the main CI

noaccOS committed:
feat: add asdf tooling

feat(nix): use tooling from tool-versions

Flake lock file updates:

• Updated input 'edgehog':
    'github:noaccOS/edgehog/c40a60e1837ccf5eb6609e209fe6bf5dee52255c?dir=backend' (2023-09-14)
  → 'github:noaccOS/edgehog/cdf5d7ac62a7900692c577150cd8446a20683b1c?dir=backend' (2023-10-05)
• Updated input 'edgehog/elixir-utils':
    'github:noaccOS/elixir-utils/0a9dccc959427d8b95625f247e9ba0c5bebe9f75' (2023-07-26)
  → 'github:noaccOS/elixir-utils/fa5b4261fd2c11b7d7ddfcb136eff139a95872a1' (2024-02-06)
• Updated input 'flake-compat':
    'github:edolstra/flake-compat/35bb57c0c8d8b62bbfd284272c928ceb64ddbde9' (2023-01-17)
  → 'github:edolstra/flake-compat/0f9255e01c2351cc7d116c072cb317785dd33b33' (2023-10-04)
• Updated input 'flake-utils':
    'github:numtide/flake-utils/ff7b65b44d01cf9ba6a71320833626af21126384' (2023-09-12)
  → 'github:numtide/flake-utils/1ef2e671c3b0c19053962c07dbda38332dcebf26' (2024-01-15)
• Updated input 'nixpkgs':
    'github:NixOS/nixpkgs/1697b7d480449b01111e352021f46e5879e47643' (2023-09-13)
  → 'github:NixOS/nixpkgs/4b1aab22192b787355733c9495d47f4c66af084c' (2024-02-05)

Signed-off-by: Francesco Noacco <francesco.noacco@secomind.com>

noaccOS commented:
> Let's use the standard CI flow here, the rest of us are not Nix-savvy enough to start using this as the main CI

reworked to use asdf instead
